src/Microsoft.Identity.Web.AgentIdentities/AgentIdentitiesExtension.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Identity.Abstractions;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Extensions methods to enable agent identities in Microsoft.Identity.Web.
+    /// </summary>
+    public static class AgentIdentityExtension
+    {
+        /// <summary>
+        /// Enable support for agent identities.
+        /// </summary>
+        /// <param name="services">Service collection</param>
+        /// <returns>the service collection for chaining</returns>
+        public static IServiceCollection AddAgentIdentities(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services), "Service collection cannot be null.");
```

Comment:
If you add this package to the Id.Wed.Diagnostics IntersalsVisibleTo file, you can do `Throws.ArgumentNullException` instead, which is a bit more performant.

Reply:
Will do. Thanks